bigtable-hbase-dataflow/src/main/java/com/google/cloud/bigtable/dataflow/HBaseImportIO.java

```diff
@@ -0,0 +1,145 @@
+
+import org.apache.hadoop.hbase.Cell;
+import org.apache.hadoop.hbase.client.Mutation;
+import org.apache.hadoop.hbase.client.Put;
+import org.apache.hadoop.hbase.client.Result;
+import org.apache.hadoop.hbase.io.ImmutableBytesWritable;
+import org.apache.hadoop.mapreduce.lib.input.SequenceFileInputFormat;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * <p>
+ * {@link com.google.cloud.dataflow.sdk.Pipeline},
+ * {@link com.google.cloud.dataflow.sdk.io.Source} and
+ * {@link com.google.cloud.dataflow.sdk.transforms.PTransform}s for importing HBase Sequence Files
+ * into <a href="https://cloud.google.com/bigtable/">Google Cloud Bigtable</a>.
+ * <p>
+ * User of this class should provide dependency for
+ * {@code org.apache.hadoop.hbase.mapreduce.ResultSerialization}, which may be found in
+ * hbase-server jars.
+ *
+ * The following code snippets sets up and runs an import pipeline:
+ *
+ * <pre>
+ * {@code
+ *   HBaseImportOptions options =
+ *       PipelineOptionsFactory.fromArgs(args).withValidation().as(HBaseImportOptions.class);
+ *   Pipeline p = CloudBigtableIO.initializeForWrite(Pipeline.create(options));
+ *   p
+ *       .apply("ReadSequenceFile", Read.from(HBaseImportIO.createSource(options)))
+ *       .apply("ConvertResultToMutations", HBaseImportIO.transformToMutations())
+ *       .apply("WriteToTable", CloudBigtableIO.writeToTable(
+ *           CloudBigtableTableConfiguration.fromCBTOptions(options)));
+ *   p.run();
+ * }
+ * </pre>
+ */
+public class HBaseImportIO {
+  private static Logger logger = LoggerFactory.getLogger(HBaseImportIO.class);
+
+  // Needed for HBase 0.94 format. Copied from ResultSerialization.IMPORT_FORMAT_VER.
+  private static final String IMPORT_FORMAT_VER = "hbase.import.version";
+  private static final String VERSION_094_STRING = "0.94";
+
+  /**
+   * Returns a {@link BoundedSource} from an HBase Sequence File for an import pipeline.
+   */
+  public static BoundedSource<
+      KV<ImmutableBytesWritable, Result>> createSource(HBaseImportOptions options) {
+    return HadoopFileSource.from(
+        options.getFilePattern(),
+        SequenceFileInputFormat.class,
+        ImmutableBytesWritable.class,
+        Result.class,
+        KvCoder.of(new WritableCoder<>(ImmutableBytesWritable.class), new HBaseResultCoder()),
+        createSerializationProperties(options.isHBase094DataFormat()));
+  }
+
+  /**
+   * <p>
+   * Returns a {@link PTransform} that converts a {@link Result} into
+   * {@link Mutation}s.
+   */
+  public static PTransform<
+      PCollection<? extends KV<ImmutableBytesWritable, Result>>,
+      PCollection<Mutation>> transformToMutations() {
+    return ParDo.of(new HBaseResultToMutationFn());
+  }
+
+
+  private static ImmutableMap<String, String> createSerializationProperties(
+      boolean isHBase094DataFormat) {
+    if (isHBase094DataFormat) {
+      return ImmutableMap.of(IMPORT_FORMAT_VER, VERSION_094_STRING);
```

Comment:
nit:
consider caching static final Map = ImmutableMap.of("hbase.import.version", "0.94");